src/CMakeLists.txt

```diff
@@ -95,9 +95,15 @@ set ( PACKAGE_VERSION_STRING "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_COMMIT
 ## Find external dependencies.
 find_package(PkgConfig)
 find_package(LibElf REQUIRED)
-find_package(hsakmt 1.0 REQUIRED HINTS ${CMAKE_INSTALL_PREFIX} PATHS /opt/rocm)
 pkg_check_modules(drm REQUIRED IMPORTED_TARGET libdrm)
 
+## Find dependencies that may be built with us or from an installed package.
+if(TARGET hsakmt::hsakmt)
+  message(STATUS "Using build dep on hsakmt")
+else()
+  find_package(hsakmt 1.0 REQUIRED HINTS ${CMAKE_INSTALL_PREFIX} PATHS /opt/rocm)
```

Comment:
The `find_package` should be disabled by the superproject either by using the `OVERRIDE_FIND_PACKAGE` in `FetchContent_Declare` or by using the `register_source_package` function as described [here](https://gitlab.kitware.com/cmake/cmake/-/issues/17735#note_487572):

```cmake
function(register_source_package NAME)
  file(WRITE "${CMAKE_BINARY_DIR}/__pkg/${NAME}/${NAME}Config.cmake" "")
  set(${NAME}_DIR ${CMAKE_BINARY_DIR}/__pkg/${NAME} CACHE PATH "")
endfunction()

```